quasarnp/io.py

```diff
@@ -295,6 +295,49 @@ def load_desi_exposure(dir_name, spec_number, fibers=np.ones(500, dtype="bool"))
     X_out = (X_out - mean) / rms
     return X_out, np.where(nonzero_weights)[0]
 
+def load_desi_coadd(filename, fibers=np.ones(500,dtype="bool")):
```

Comment:
I added a doc string to load_desi_coadd(), but, since I wrote it, no one has used it yet but me, thankfully. Running off targetids might be easier. @dylanagreen and @dkirkby, would this have a downstream effect on efficiency?

Reply:
I don't think running off targetids should have a downstream effect on efficiency. It's probably not _much_ slower than running off fibers and taking an array of fibers to keep. Most of the overhead would come in matching the target ids with the loaded data to know which entries to keep and return. Most implementations of this use astropy.table and an inner join, but we'd have to do it differently in order to avoid including the rather heavyweight astropy dependency.

Reply:
to avoid the astropy dependency while filtering on TARGETID, I suggest using `numpy.in1d`, e.g.
```
with fitsio.FITS(filename) as h:
    fibermap = h['FIBERMAP'].read()
    if targetids is not None:
        keep = np.in1d(fibermap['TARGETID'], targetids)
    else:
        keep = np.ones(len(fibermap), bool)

    #- proceed with the rest of the code, using "keep" instead of original "fibers", e.g.
    flux = h[fluxname].read()[keep, :]
```

Reply:
Yes, I prefer not to add any new dependencies to keep this package lightweight.

Note that the [numpy docs](https://numpy.org/doc/stable/reference/generated/numpy.in1d.html) recommend using `np.isin` instead of `np.in1d`.

Reply:
I renamed the argument "rows" to indicate that it isn't using fibers and reset the default value to None. If it is none, it now pulls the number of rows in the coadd file to use as "nfibers". The user can instead pass it a boolean array of which rows to use (as they could before) and it will use that instead.

As for sharing code between load_desi_coadd() and load_desi_exposure(), I am not comfortable changing the exposure function for the same reasons we didn't want a "cbest" output, it would change other hooks and compatibility with QuasarNET.